[Outlook] (manifest) relation of module extension to JSON manifest (#3609)
OfficeDev/office-js-docs-pr: docs/outlook/extension-module-outlook-add-ins.md

---
title: Module extension Outlook add-ins
description: Create applications that run inside Outlook to make it easy for your users to access business information and productivity tools without ever leaving Outlook.
ms.date: 08/30/2022
ms.localizationpriority: medium
---

# Module extension Outlook add-ins

Module extension add-ins appear in the Outlook navigation bar, right alongside mail, tasks, and calendars. A module extension is not limited to using mail and appointment information. You can create applications that run inside Outlook to make it easy for your users to access business information and productivity tools without ever leaving Outlook.

> [!TIP]

> > Module extensions aren't supported in the [Teams manifest (preview)](../develop/json-manifest-overview.md), but you can create a very similar experience for users by making a [personal tab that opens in Outlook](/microsoftteams/platform/m365-apps/extend-m365-teams-personal-tab). In the early preview period for the Teams manifest in Outlook Add-ins, it is not possible to combine an Outlook Add-in and a personal tab in the same manifest and install them as a unit. We're working on this but in the meantime you must create separate apps for the add-in and the personal tab. They can both use files on the same domain.

Reviewer: ```suggestion
> Module extensions aren't supported in the [Teams manifest (preview)](../develop/json-manifest-overview.md), but you can create a very similar experience for users by making a [personal tab that opens in Outlook](/microsoftteams/platform/m365-apps/extend-m365-teams-personal-tab). In the early preview period for the Teams manifest in Outlook Add-ins, it isn't possible to combine an Outlook Add-in and a personal tab in the same manifest and install them as a unit. We're working on this, but in the meantime, you must create separate apps for the add-in and the personal tab. They can both use files on the same domain.
```

![Shows the expanded navigation bar when more than on module extension is loaded in Outlook.](../images/outlook-module-navigationbar-more.png)

> When you click on an extension, Outlook replaces the built-in module with your custom module so that your users can interact with the add-in. You can use some of the features of the Outlook JavaScript API in your add-in. APIs that logically assume a specific Outlook item such as a message or appointment don't work in module extensions. The module can also include function commands in the Outlook ribbon that interact with the add-in's page. To facilitate this, your function commands call the [Office.onReady or Office.initialize method](../develop/initialize-add-in.md) and the [Event.completed](/javascript/api/office/office.addincommands.event#office-office-addincommands-event-completed-member(1)) method. See the sample [Outlook module extensions Billable hours sample](https://github.com/OfficeDev/Outlook-Add-in-JavaScript-ModuleExtension).

Reviewer: ```suggestion
When you click on an extension, Outlook replaces the built-in module with your custom module so that your users can interact with the add-in. You can use some of the features of the Outlook JavaScript API in your add-in. APIs that logically assume a specific Outlook item, such as a message or appointment, don't work in module extensions. The module can also include function commands in the Outlook ribbon that interact with the add-in's page. To facilitate this, your function commands call the [Office.onReady or Office.initialize method](../develop/initialize-add-in.md) and the [Event.completed](/javascript/api/office/office.addincommands.event#office-office-addincommands-event-completed-member(1)) method. To walk through how a module extension Outlook add-in is configured, see the [Outlook module extensions billable hours sample](https://github.com/OfficeDev/Outlook-Add-in-JavaScript-ModuleExtension).
```
